services/sonar/sonar-tests.tester.js

```diff
@@ -100,7 +100,12 @@ t.create('Test Failures Count')
   )
   .expectBadge({
     label: 'test failures',
-    message: Joi.alternatives(isMetric, 0),
+    message: Joi.alternatives(
+      isMetric,
+      Joi.number()
+        .valid(0)
+        .required()
+    ),
```

Comment:
This rather log schema is repeated a few times in this file. Would you mind refactoring it?

Reply:
I had the same thought (although I also thought I should have done that to begin with when I wrote the original 😄)